Support importing Adium logs

gregkare opened the pull request:
Connected to #2

gregkare committed:
Update remotestoragejs library, use remotestorage-module-chat-messages

Also do not use babel anymore

gregkare committed:
Add initial Adium importer

raucao commented on importers/adium.js:
```diff
@@ -0,0 +1,222 @@
+  var mucServer = program.adiumMuc;
+
+  if (!fs.existsSync(dir)) {
+    console.log('Input directory doesn\'t contain Adium logs (should be in ~/Library/Application\ Support/Adium\ 2.0/Users/Default/Logs/Jabber.[adium-user]/)');
```
Not sure if the mere existence of the directory confirms that there are Adium logs in it?

raucao commented on importers/adium.js:
```diff
@@ -0,0 +1,222 @@
+    //     var r = new RegExp(program.zncUser+'\/'+program.zncNetwork+'\/'+program.zncChannel+'/', 'i');
+    //     return filename.match(r);
+    //   });
+    // }
```
These commented sections seem to be obsolete.

raucao commented on importers/adium.js:
```diff
@@ -0,0 +1,222 @@
+  pending.resolve();
+
+  remoteStorage.access.claim("messages-irc", "rw");
+  // remoteStorage.caching.disable("/");
```
Sure this is supposed to use caching? I don't quite see why it would.

raucao commented on importers/adium.js:
```diff
@@ -0,0 +1,222 @@
+
+  if (!fs.existsSync(program.input)) {
+    console.error('Input directory doesn\'t exist');
+    process.exit(1);
```
This same check is in line 16? Looks like only one of the two is needed.

raucao commented:
Added some comments.

gregkare committed:
Remove sanity check that is already done above

gregkare committed:
Remove commented leftovers from the znc adapter

gregkare committed:
Disable caching when instantiating RemoteStorage

raucao commented on importers/adium.js:
```diff
@@ -9,7 +9,7 @@ var parseString = require('xml2js').parseString;
 var RemoteStorage = require("remotestoragejs");
 var ChatMessages = require("remotestorage-module-chat-messages");
-var remoteStorage = new RemoteStorage({ modules: [ ChatMessages.default ] });
+var remoteStorage = new RemoteStorage({ caching: false, modules: [ ChatMessages.default ] });
```
According to https://remotestoragejs.readthedocs.io/en/latest/js-api/remotestorage.html it's `cache: false`.

raucao commented on importers/znc.js:
```diff
@@ -7,7 +7,7 @@ var program     = null;
 var RemoteStorage = require("remotestoragejs");
 var ChatMessages = require("remotestorage-module-chat-messages");
-var remoteStorage = new RemoteStorage({ modules: [ ChatMessages.default ] });
+var remoteStorage = new RemoteStorage({ caching: false, modules: [ ChatMessages.default ] });
```
See above.

gregkare committed:
Fix the constructor's cache config

gregkare commented:
@skddc I'm getting an issue with the `npm run importer` command, just checking if you've seen that before I investigate:

When I run it like this:

```
$ npm run importer -t adium -i /Users/kare/Library/Application\ Support/Adium\ 2.0/Users/Default/Logs/Jabber.greg@test.com --rs-user test --rs-token test
```

The switch arguments get lost: 

> rs-messages-importer@0.5.0 importer /Users/kare/code/kosmos/rs-messages-importer
> node index.js "adium" "/Users/kare/Library/Application Support/Adium 2.0/Users/Default/Logs/Jabber.greg@5apps.com" "test" "test"


I'm getting the same issue with the master branch, so it might be that npm changed the way `npm run` behaves

gregkare committed:
Do not resolve the promise before connecting to the RS account

This was leftover from a temporary version of the code where I didn't
connect RS for debugging purposes

raucao commented:
This is standard npm behaviour. You need to put any arguments to the script behind `--`.

Edit:

```bash
~ $ npm run --help
npm run-script <command> [-- <args>...]
```

gregkare commented:
Got it, the README's example only uses one dash (that doesn't work), that's what confused me

gregkare commented:
I'm fixing a small bug that's losing links in the middle of messages, then I'll compare my imported logs to the ones that botka has been logging to make sure everything works

gregkare committed:
Merge multiple log files from the same day, support links in text

gregkare commented:
I just pushed a fix for links in the middle of text and now support merging multiple log files for the same day (Adium creates a new log file for every session, not one per day).

I still need to fix an issue with nested XML, for example this line which isn't logged (pasting so I can easily find it again):

```
<message sender="sebastian@5apps.com" time="2017-10-30T13:53:20+0100" alias="basti"><div><span style="color: ##204a87;"><span style="font-family: Helvetica; font-size: 12pt;">(01:51:27 PM) </span><span style="font-family: Helvetica;"><span style="font-size: 14pt; font-weight: bold;">basti: </span></span></span><span style="font-family: Helvetica; font-size: 14pt;">is that maybe bitgo-express?</span><br /><span style="color: ##48c4e8;"><span style="font-family: Helvetica; font-size: 12pt;">(01:52:01 PM) </span><span style="font-family: Helvetica;"><span style="font-size: 14pt;">bumi: </span></span></span><span style="font-family: Helvetica; font-size: 14pt;">yep</span></div></message>
```

gregkare committed:
Support messages that were logged as nested spans (any number of levels)

gregkare commented:
Some messages are going to the wrong day, fixing it

gregkare committed:
Fix some messages being sent to the wrong day

It happened when the first message from a log file was received the day
before

gregkare committed:
Use the same order of attributes as in the hubot logger

gregkare commented:
Now fixing another bug with multiline messages, almost there

gregkare committed:
Correctly import Adium messages that span multiple lines

gregkare commented:
I just pushed my fix for messages that are multiple lines

gregkare committed:
Fix the error message when the input directory does not exist

gregkare commented:
I have also fixed the error message when the input directory does not exist. This should be ready to review/merge

gregkare committed:
Add missing links from secondary spans, and missing space after links

gregkare commented:
I found two more small issues with links and fixed them in the commit above

gregkare committed:
Use aliases instead of full XMPP identifiers, like the hubot logger

gregkare committed:
Fix importing messages with nested spans

gregkare commented:
I added another fix for some messages, I'm not so good with recursion so it's probably possible to improve the code a lot
